lib/cadet/accounts/notification.ex

```diff
@@ -189,4 +193,116 @@ defmodule Cadet.Accounts.Notification do
       assessment_id: assessment_id
     })
   end
+
+  @doc """
+  Writes a notification that a student's submission has been
+  autograded successfully. (for the student)
+  """
+  @spec write_notification_when_autograded(integer() | String.t()) :: Ecto.Changeset.t()
+  def write_notification_when_autograded(submission_id) do
+    submission =
+      Submission
+      |> Repo.get_by(id: submission_id)
+
+    params = %{
+      type: :autograded,
+      read: false,
+      role: :student,
+      user_id: submission.student_id,
+      assessment_id: submission.assessment_id,
+      submission_id: submission_id
+    }
+
+    write(params)
+  end
+
+  @doc """
+  Writes a notification that a student's submission has been
+  manually graded successfully by an Avenger or other teaching staff.
+  (for the student)
+  """
+  @spec write_notification_when_manually_graded(integer() | String.t()) :: Ecto.Changeset.t()
+  def write_notification_when_manually_graded(submission_id) do
```

Comment:
These 2 functions are similar, we can combine them together into `write_notification_when_graded(submission_id, type) with type in [:graded, :autograded] do`